src/main/java/flash/npcmod/client/gui/screen/TreeBuilderScreen.java

```diff
@@ -594,31 +681,28 @@ public void setNodeBeingEdited(@Nullable BuilderNode node, EditType editType) {
             this.functionParamsField.setValue("");
         }
         boolean check;
-        for (EditType key : allFields.keySet()) {
+        this.fileNameFrame.setVisible(editType == EditType.FILENAME);
+        this.nodeNameFrame.setVisible(editType == EditType.NAME);
+
+        for (EditType key : this.allNameFields.keySet()) {
             check = editType == key;
-            allFields.get(key).setVisible(check);
-            allFields.get(key).setFocus(check);
-            if (isNodeNull) allFields.get(key).setValue("");
+            this.allNameFields.get(key).setFocus(check);
+            this.allTopLevelFrames.get(key).setVisible(check);
```

Comment:
Please do null checks on stuff like this :)

```
java.lang.NullPointerException: Cannot invoke "flash.npcmod.client.gui.widget.DirectionalFrame.setVisible(boolean)" because the return value of "java.util.HashMap.get(Object)" is null
	at flash.npcmod.client.gui.screen.TreeBuilderScreen.setNodeBeingEdited(TreeBuilderScreen.java:690) ~[%2384!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at flash.npcmod.client.gui.screen.dialogue.DialogueBuilderScreen.setNodeBeingEdited(DialogueBuilderScreen.java:303) ~[%2384!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at flash.npcmod.client.gui.dialogue.DialogueNode.clickedOnTextBar(DialogueNode.java:198) ~[%2384!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at flash.npcmod.client.gui.dialogue.DialogueNode.clickedOn(DialogueNode.java:102) ~[%2384!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at flash.npcmod.client.gui.screen.dialogue.DialogueBuilderScreen.lambda$mouseClicked$0(DialogueBuilderScreen.java:201) ~[%2384!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?] {}
	at flash.npcmod.client.gui.screen.dialogue.DialogueBuilderScreen.mouseClicked(DialogueBuilderScreen.java:201) ~[%2384!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.MouseHandler.lambda$onPress$0(MouseHandler.java:88) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screens.Screen.wrapScreenError(Screen.java:528) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.MouseHandler.onPress(MouseHandler.java:85) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.MouseHandler.lambda$setup$4(MouseHandler.java:185) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.util.thread.BlockableEventLoop.execute(BlockableEventLoop.java:90) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.client.MouseHandler.lambda$setup$5(MouseHandler.java:184) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36) ~[lwjgl-glfw-3.2.2.jar%2357!/:build 10] {}
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.2.2.jar%2363!/:build 10] {}
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3101) ~[lwjgl-glfw-3.2.2.jar%2357!/:build 10] {}
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:164) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at com.mojang.blaze3d.platform.Window.updateDisplay(Window.java:333) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.runTick(Minecraft.java:1067) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.run(Minecraft.java:664) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:205) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.fml.loading.targets.ForgeClientUserdevLaunchHandler.lambda$launchService$0(ForgeClientUserdevLaunchHandler.java:24) ~[fmlloader-1.18.2-40.1.68.jar%230!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:149) [bootstraplauncher-1.0.0.jar:?] {}
```